addons/mail/static/src/components/chat_window_manager/chat_window_manager_tests.js

```diff
@@ -725,59 +705,100 @@ QUnit.test('chat window: state conservation on toggle home menu', async function
-    assert.strictEqual(
-        document.querySelector(`.o_ThreadViewer_messageList`).scrollTop,
-        142,
-        "chat window scrollTop should still be the same (1)"
-    );
-    composerTextInputTextArea = document.querySelector(`.o_ComposerTextInput_textarea`);
+
+    await this.hideHomeMenu();
     assert.strictEqual(
-        composerTextInputTextArea.value,
+        document.querySelector(`.o_ComposerTextInput_textarea`).value,
         "XDU for the win !",
         "Chat window composer should still have the same html input (1)"
     );
-    assert.strictEqual(
-        document.querySelectorAll(`.o_Composer .o_Attachment`).length,
+    assert.containsN(
+        document.body,
+        '.o_Composer .o_Attachment',
         2,
         "Chat window composer should have 2 attachments (1)"
     );
 
     // Show home menu
-    await this.env.bus.trigger('will_show_home_menu');
-    await this.env.bus.trigger('show_home_menu');
+    await this.showHomeMenu();
+    assert.strictEqual(
+        document.querySelector(`.o_ComposerTextInput_textarea`).value,
+        "XDU for the win !",
+        "chat window composer should still have the same html input (2)"
+    );
+    assert.containsN(
+        document.body,
+        '.o_Composer .o_Attachment',
+        2,
+        "Chat window composer should have 2 attachments (2)"
+    );
+});
+
+QUnit.test('[technical] chat window: scroll conservation on toggle home menu', async function (assert) {
+    assert.expect(3);
+
+    Object.assign(this.data.initMessaging, {
+        channel_slots: {
+            channel_channel: [{
+                channel_type: 'channel',
+                id: 20,
+                is_minimized: false,
+                is_pinned: true,
+                name: "General",
+                state: 'open',
+                uuid: 'channel-20-uuid',
+            }],
+        },
+    });
+    this.data['mail.message'].records = [];
+    for (const i of Array(20).keys()) {
+        this.data['mail.message'].records.push({
+            author_id: [7, "Demo"],
+            body: "<p>body</p>",
+            channel_ids: [20],
+            date: "2019-04-20 10:00:00",
+            id: i + 10,
+            message_type: 'comment',
+            model: 'mail.channel',
+            record_name: 'General',
+            res_id: 20,
+        });
+    }
+    await this.start();
+    await afterNextRender(() => document.querySelector(`.o_MessagingMenu_toggler`).click());
+    await afterNextRender(() =>
+        document.querySelector(`.o_MessagingMenu_dropdownMenu .o_NotificationList_preview`).click()
+    );
+    // Set a scroll position to chat window
+    document.querySelector(`.o_ThreadViewer_messageList`).scrollTop = 142;
     assert.strictEqual(
         document.querySelector(`.o_ThreadViewer_messageList`).scrollTop,
         142,
-        "chat window scrollTop should still be the same (2)"
+        "verify chat window initial scrollTop"
```

Comment:
weird to formulate assertion comment with "verify X". Prefer expected (explicit) outcome with modal verb, like "should have scrolled to X"

Reply:
Not resolved